### ISSUE TITLE
Object.defineProperty proxy trap

### ISSUE THREAD

**Issue by dcamilleri** [sdgluck/mewt#15]
Hi, 

To enhance object immutability, it would be nice to implement a proxy trap for `Object.defineProperty`. 

Example: 

```js
// Current version 
const o = mewt({})
Object.defineProperty(o, 'foo', {
     value: 'bar'
})

o.foo // 'bar' => we expect an error here
```

What do you think about adding this new trap ? If you want to add it, I've got the code ready on my fork, I can send a PR.

**Comment by sscotth** [comment]
I looked at your implementation. If you use a `defineProperty` handler, I don't think you need a `set` handler.

https://ponyfoo.com/articles/es6-proxy-traps-in-depth#it-s-a-trap

**Comment by sdgluck** [comment]
@dcamilleri Thank you for suggestion. I definitely would like to include this, taking into account @sscotth's comments. 👍 If you're willing to update your fork and create a PR, I will review.

Also heads up that on your branch the function signature is now in the wrong place.

**Comment by bali182** [comment]
@sscotth Can we return the newly created clone with the new key - value from `defineProperty`? Docs says it should return a `boolean` not sure what happens with the return value then.

**Comment by bali182** [comment]
If we want it really short, wouldn't something like this be enough to prevent all mutations?

```js
const mutationTrap = () => throw new Error(`${isA ? 'array' : 'object'} is immutable`)
['setPrototypeOf', 'defineProperty', 'set', 'deleteProperty']
  .reduce((handler, key) => { handler[key] = mutationTrap; return handler }, handler)
```
Where handler is the object we are passing to the `Proxy` constructor?

**Comment by sdgluck** [comment]
Fixed in #21
